plaidbench.py

```diff
@@ -257,79 +260,86 @@ def main():
-
-        # write all data to result.json / report.npy if single run
-        else:
-            network_data['example'] = network
+            # Run network w/ batch_size
             try:
-                os.makedirs(args.result)
-            except OSError as ex:
-                if ex.errno != errno.EEXIST:
-                    printf(ex)
-                    return
-            with open(os.path.join(args.result, 'result.json'), 'w') as out:
-                json.dump(network_data, out)
-            if isinstance(output.contents, np.ndarray):
-                np.save(os.path.join(args.result, 'result.npy'), output.contents)
-            # close
-            sys.exit(exit_status)
+                value_check(examples, epochs, batch_size)
+
+                # Setup
+                x_train, y_train = setup(args.train, epoch_size, batch_size)
+
+                # Loading the model
+                module, x_train, model = load_model(args.module, x_train)
+
+                if args.train:
+                    # training run
+                    train(x_train, y_train, epoch_size, model, batch_size, 
+                          compile_stop_watch, epochs, stop_watch, output, network)
+                else:
+                    # inference run
+                    inference(args.module, model, batch_size, compile_stop_watch, 
+                              output, x_train, examples, stop_watch)
+
+                # Record stopwatch times
+                execution_duration = stop_watch.elapsed()
+                compile_duration = compile_stop_watch.elapsed()
+
+                network_data['compile_duration'] = compile_duration
+                network_data['execution_duration'] = execution_duration / examples
+                network_data['precision'] = output.precision
+                network_data['example_size'] = examples
+                network_data['batch_size'] = batch_size
+
+                # Print statement
+                printf('Example finished, elapsed: {} (compile), {} (execution)'.format(
+                    compile_duration, execution_duration))
+
+            # Error handling
+            except Exception as ex:
+                # Print statements
+                printf(ex)
+                printf('Set --print-stacktraces to see the entire traceback')
+
+                # Record error
+                network_data['exception'] = str(ex)
+                
+                # Set new exist status
+                exit_status = -1
+
+                # stacktrace loop
+                if args.print_stacktraces:
+                    raise NotImplementedError                
+
+            # stores network data in dictionary
+            if args.blanket_run:
+                composite_str = network + " : " + str(batch_size)
+                outputs[composite_str] = dict(network_data)
+            # write all data to result.json / report.npy if single run
+            else:
+                network_data['model'] = network
```

Comment:
move this above where all the other network_data settings are (i.e., set this for blanket too)